specification/apicenter/resource-manager/Microsoft.ApiCenter/preview/2024-03-15-preview/apicenter.json

```diff
@@ -0,0 +1,4985 @@
+          },
+          {
+            "name": "system",
+            "value": "system",
+            "description": "Origin with System"
+          },
+          {
+            "name": "user,system",
+            "value": "user,system",
+            "description": "Origin with user and system"
+          }
+        ]
+      }
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "description": "The provisioning state of the resource",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Succeeded",
+            "value": "Succeeded",
+            "description": "Resource has been created."
+          },
+          {
+            "name": "Failed",
+            "value": "Failed",
+            "description": "Resource creation failed."
+          },
+          {
+            "name": "Canceled",
+            "value": "Canceled",
+            "description": "Resource creation was canceled."
+          }
+        ]
+      },
+      "readOnly": true
+    },
+    "Service": {
+      "type": "object",
+      "description": "The service entity.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ServiceProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity",
+          "description": "The managed service identities assigned to this resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "ServiceListResult": {
+      "type": "object",
+      "description": "The response of a Service list operation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The Service items on this page",
+          "items": {
+            "$ref": "#/definitions/Service"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "ServiceProperties": {
+      "type": "object",
+      "description": "The properties of the service.",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "Provisioning state of the service.",
+          "readOnly": true
+        },
+        "restore": {
```

Comment:
consider using enum instead of boolean:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md

Reply:
@ms-zhenhua Looking at the documentation that you linked, I do think a `boolean` is more appropriate here. 

In the document it says:

>... the intent is to consider if there could be more than 2 values possible for the property in the future or not. If the answer is no, then a boolean is fine, if the answer is yes, there could be other values added in the future, making it an enum can help avoid breaking changes in the SDKs in the future.

Restore can _only_ be _two_ value, either the customer wants to restore an APIC instance (`restore = true`) or the customer wants to create a new APIC instance (`restore=false`), there is no in between. Therefore. I think `boolean` is appropriate in this case.

Reply:
Agree with @alexyaang here.